SPECS/azurelinux-release/azurelinux-release.spec

```
@@ -1,67 +1,125 @@
+Source3:        99-default-disable.preset
+
+Provides:       system-release
+Provides:       system-release(%{version})
+
+Conflicts:      mariner-rpm-macros < 2.0-25
+
 BuildArch:      noarch
 
+BuildRequires:  systemd-rpm-macros
+
 %description
-Azure Linux release files such as yum configs and other %{_sysconfdir}/ release related files
+Azure Linux release files such as dnf configs and other %{_sysconfdir}/ release related files
+and systemd preset files that determine which services are enabled by default.
 
 %install
 install -d %{buildroot}%{_sysconfdir}
-install -d %{buildroot}/%{_libdir}
+install -d %{buildroot}%{_libdir}
 
-echo "Azure Linux %{mariner_release_version}" > %{buildroot}%{_sysconfdir}/azurelinux-release
-echo "AZURELINUX_BUILD_NUMBER=%{mariner_build_number}" >> %{buildroot}%{_sysconfdir}/azurelinux-release
+cat <<-"EOF" > %{buildroot}%{_libdir}/azurelinux-release
+%{distribution} %{version}
+AZURELINUX_BUILD_NUMBER=%{mariner_build_number}
+EOF
+ln -sv ..%{_libdir}/azurelinux-release %{buildroot}%{_sysconfdir}/azurelinux-release
 
-cat > %{buildroot}%{_sysconfdir}/lsb-release <<- "EOF"
+cat <<-"EOF" > %{buildroot}%{_libdir}/lsb-release
 DISTRIB_ID="azurelinux"
-DISTRIB_RELEASE="%{mariner_release_version}"
+DISTRIB_RELEASE="%{version}"
 DISTRIB_CODENAME=AzureLinux
-DISTRIB_DESCRIPTION="Microsoft Azure Linux %{mariner_release_version}"
+DISTRIB_DESCRIPTION="%{distribution} %{version}"
 EOF
+ln -sv ..%{_libdir}/lsb-release %{buildroot}%{_sysconfdir}/lsb-release
 
-version_id=`echo %{mariner_release_version} | grep -o -E '[0-9]+.[0-9]+' | head -1`
-cat > %{buildroot}/%{_libdir}/os-release << EOF
-NAME="Microsoft Azure Linux"
-VERSION="%{mariner_release_version}"
+cat <<-"EOF" > os-release
+NAME="%{distribution}"
+VERSION="%{version}"
 ID=azurelinux
-VERSION_ID="$version_id"
-PRETTY_NAME="Microsoft Azure Linux $version_id"
+VERSION_ID="%{version}"
+PRETTY_NAME="%{distribution} %{version}"
 ANSI_COLOR="1;34"
 HOME_URL="%{url}"
 BUG_REPORT_URL="%{url}"
 SUPPORT_URL="%{url}"
 EOF
+ln -sv ..%{_libdir}/os-release %{buildroot}%{_sysconfdir}/os-release
 
-ln -sv ../usr/lib/os-release %{buildroot}%{_sysconfdir}/os-release
-
-cat > %{buildroot}%{_sysconfdir}/issue <<- EOF
-Welcome to Azure Linux %{mariner_release_version} (%{_arch}) - (\l)
+cat <<-"EOF" > %{buildroot}%{_libdir}/issue
+Welcome to %{distribution} %{version} (%{_arch}) - (\l)
 EOF
+ln -sv ..%{_libdir}/issue %{buildroot}%{_sysconfdir}/issue
 
-cat > %{buildroot}%{_sysconfdir}/issue.net <<- EOF
-Welcome to Azure Linux %{mariner_release_version} (%{_arch})
+cat <<-"EOF" > %{buildroot}%{_libdir}/issue.net
+Welcome to %{distribution} %{version} (%{_arch})
+EOF
+ln -sv ..%{_libdir}/issue.net %{buildroot}%{_sysconfdir}/issue.net
+
+install -d -m 755 %{buildroot}%{_sysconfdir}/issue.d
+
+cat <<-"EOF" > %{buildroot}%{_rpmmacrodir}/macros.dist
+# dist macros.
+
+%%__bootstrap         ~bootstrap
+%%azure               %{dist_version}
+%%azl%{dist_version}  1
```

We decided that `azl<ver> 1` does not give much value over `azl <ver>`
```suggestion
%%azl                %{dist_version}
```